api/types/common/common.go

```diff
@@ -16,22 +16,43 @@ limitations under the License.
 
 package common
 
-import "regexp"
+// IsValidLabelKey checks if the supplied string matches the
+// label key regexp.
+func IsValidLabelKey(s string) bool {
+	for _, b := range []byte(s) {
+		if !isValidLabelKeyByte(b) {
+			return false
+		}
+	}
+	return true
+}
 
-// LabelPattern is a regexp that describes a valid label key. In this case,
-// valid label keys can consist of alphanumeric characters, forward slashes,
+// isValidLabelKeyByte checks if the byte is a valid character for a label key.
+// Valid label keys can consist of alphanumeric characters, forward slashes,
 // periods, underscores, colons, stars, and dashes. Some valid label examples:
 //
 // label
 // teleport.dev/fine-grained-access
 // teleport.dev/managed:internal_access
 // all-objects*
-const LabelPattern = `^[a-zA-Z/.0-9_:*-]+$`
-
-var validLabelKey = regexp.MustCompile(LabelPattern)
-
-// IsValidLabelKey checks if the supplied string matches the
-// label key regexp.
-func IsValidLabelKey(s string) bool {
-	return validLabelKey.MatchString(s)
+func isValidLabelKeyByte(b byte) bool {
+	switch b {
+	case
+		// Digits
+		'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+
+		// Lowercase letters
+		'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+		'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+
+		// Uppercase letters
+		'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+		'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+
+		// Allowed symbols
+		'/', '.', '_', ':', '*', '-':
+		return true
+	default:
+		return false
+	}
```

Comment:
nit: don't have strong opinion: 
```suggestion
func isValidLabelKeyByte(b byte) bool {
	return ('0' <= b && b <= '9') || // Digits
		('a' <= b && b <= 'z') || // Lowercase letters
		('A' <= b && b <= 'Z') || // Uppercase letters
		b == '/' || b == '.' || b == '_' || b == ':' || b == '*' || b == '-' // Allowed symbols
}
```

Reply:
On my cpu this is actually a tiny bit faster (82ns/op vs 89ns/op) but it depends on the data (it becomes worse after adding another test case that has a lot of symbols), I think that the big switch table gives the compiler the best chance to optimize the code now and in the future with better bitwise and numerical tricks (that being said, the chain of conditions and the switch are equivalent so a Sufficiently Smart Compiler should just generate the exact same code for both, but that's basically a pipe dream for Go unfortunately).

I also prefer the aesthetic of the table of letters and numbers. 😌

Reply:
I'm fine with both
I'll keep it as is, but I'm happy to follow up on this